src/functions/meta-assistant/send-daily-message/handler.ts

```diff
@@ -1,67 +1,87 @@
+      return slackUsers.find(slackUser => slackUser.profile.first_name === firstName && slackUser.profile.last_name === lastName);
+    };
 
-    if (!timebankUsers) {
-      throw new Error("No persons retrieved from Timebank");
-    }
+    // Combine user data
+    const combinedUserData: DailyCombinedData[] = await Promise.all(
+      Array.isArray(filteredSeveraUsers) ? filteredSeveraUsers.map(async user => {
+        const userWorkHours = Array.isArray(workHours) ? workHours.filter(hour => hour.user.guid === user.guid) : [];
+        const workDays = await severaApi.getWorkDays(user.guid);
+        const quantity = userWorkHours[0]?.quantity || 0;
+        const enteredHours = workDays[0]?.enteredHours || 0;
+        const expectedHours = user.workContract.dailyHours || 0;
+        const minimumBillableRate = 75;
 
-    const dailyEntries: DailyEntry[] = [];
+        let totalBillableTime = 0;
 
-    for (const timebankUser of timebankUsers) {
-      const dailyEntry = await TimeBankApiProvider.getDailyEntries(timebankUser.id, yesterday, yesterday, accessToken);
-      if (dailyEntry && !dailyEntry.isVacation) {
-        dailyEntries.push(dailyEntry);
-      }
-    }
+        Array.isArray(workHours) && workHours.forEach(hour => {
+          if (hour.isBillable && hour.user.guid === user.guid) {
+            totalBillableTime += hour.quantity;
+          }
+        });
 
-    const filteredTimebankUsers = timebankUsers.filter(person => dailyEntries.find(dailyEntry => dailyEntry.person === person.id));
+        const nonBillableProject = enteredHours - totalBillableTime;
 
-    const dailyCombinedData = TimebankUtilities.combineDailyData(filteredTimebankUsers, dailyEntries, slackUsers);
-    const messagesSent = await SlackUtilities.postDailyMessageToUsers(dailyCombinedData, timeRegistrations, previousWorkDays, nonProjectTimes);
+        // Find the corresponding Slack user
+        const slackUser = findSlackUser(user.firstName, user.lastName);
 
-    const errors = messagesSent.filter(messageSent => messageSent.response.error);
+        const result = {
+          userGuid: user.guid,
+          firstName: user.firstName,
+          lastName: user.lastName,
+          enteredHours: enteredHours,
+          expectedHours: expectedHours,
+          quantity: quantity,
+          minimumBillableRate: minimumBillableRate,
+          totalBillableTime: totalBillableTime,
+          nonBillableProject: nonBillableProject,
+          date: previousWorkDays.yesterday.toISO(),
+          slackId: slackUser ? slackUser.id : null
+        };
 
-    if (errors.length) {
-      let errorMessage = "Error while posting slack messages, ";
+        return result;
+      }) : []
+    );
 
-      errors.forEach(error => {
-        errorMessage += `${error.response.error}\n`;
-      });
-      console.error(errorMessage);
-    }
+    const messageSent = await SlackUtilities.postDailyMessageToUsers(combinedUserData, previousWorkDays);
```

Comment:
Should we need this console.log here in develop branch, if not , this could be removed